### PR TITLE
Add JNI bridge for setup_payload

### LINKED ISSUE #1245: Create JNI layer for libSetupPayload
Create a JNI layer around libCHIP, providing access to setup_payload and device connection manager in prep for qr cord parsing, echo request and on/off request from Android app

### PR DESCRIPTION
**Pull request by vidhis88**
Adding the JNI bridging C++ and Java code to setup_payload package for use in Android apps.

### CODE REVIEW

File: src/inet/InetInterface.cpp [hunk 1]
```diff
@@ -224,6 +224,133 @@ void CloseIOCTLSocket(void)
...
 
+#if __ANDROID__ && __ANDROID_API__ < 24
+
+static struct if_nameindex * backport_if_nameindex(void);
+static void backport_if_freenameindex(struct if_nameindex *);
+
+static void backport_if_freenameindex(struct if_nameindex * inArray)
+{
+    if (inArray == NULL)
+        return;
```
Comment (BroderickCarlin) [inline src/inet/InetInterface.cpp:235]: Please use curly brackets for all `if()` statements

File: src/setup_payload/java/Makefile.am [hunk 2]
```diff
@@ -0,0 +1,106 @@
+#
+#    Copyright (c) 2014-2017 Nest Labs, Inc.
+#    Copyright (c) 2018 Google LLC
+#	 Copyright (c) 2020 Project CHIP Authors
```
Comment (BroderickCarlin) [inline src/setup_payload/java/Makefile.am:4]: nit: indentation is a bit off

File: src/inet/InetInterface.cpp (hunk 1 above)
Comment (vidhis88) [reply src/inet/InetInterface.cpp]: Done.

File: src/setup_payload/java/Makefile.am (hunk 2 above)
Comment (vidhis88) [reply src/setup_payload/java/Makefile.am]: Done.

**Review by BroderickCarlin — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Review by woody-apple — APPROVED** [review]

**Pull request merged**